Adds Ketrazine, a powerful combat stimulant with severe side-effects which can be synthesized by the crew

Ordonis opened the pull request:
:cl: Ordo
rscadd: Adds Ketrazine, a powerful but dangerous combat stim which the crew can synthesize
/:cl:

Ordonis committed:
Adds a Recipe for morphine to medicine

Exactly what it says on the tin.

Ordonis committed:
Merge remote-tracking branch 'refs/remotes/tgstation/master' into Recipe-updates

Ordonis committed:
Merge remote-tracking branch 'refs/remotes/tgstation/master' into Recipe-updates

Ordonis committed:
Ketrazine

Adds ketrazine, a combat stimulant with severe consequences for use.

Ordonis committed:
Merge remote-tracking branch 'refs/remotes/tgstation/master' into Recipe-updates

Ordonis committed:
Revert "Ketrazine"

This reverts commit d618fe337cc28f5513c59ff3399894c10dde4a7d.

Ordonis committed:
Revert "Revert "Ketrazine""

This reverts commit 3a4d507962874bff855ca778548bad301404471a.

coiax commented on code/modules/reagents/chemistry/reagents/medicine_reagents.dm:
```diff
@@ -1105,3 +1105,107 @@
+			M.adjustStaminaLoss(40*REM,0)
+			M.losebreath +=1
+
+			=1
```
=1?

Ordonis replied:
Is that not required at the end of a (switch)? I was under the impression that it was.

coiax commented on code/modules/reagents/chemistry/recipes/medicine.dm:
```diff
@@ -226,3 +226,18 @@
+	id = "morphine"
+	results = list("morphine" = 3)
+	required_reagents = list("carbon" = 2, "hydrogen" = 2, "nitrogen" = 1, "oxygen" = 1)
+	required_temp = 480
```
Did you mean to include this?

Ordonis replied:
No, I am a double dumbfuck for first pushing this entire thing to the morphine PR, and now for leaving the morphine thing in this, apologies

Ordonis committed:
Atomizes PRs and removes offending =1

On the tin.

GuyonBroadway commented:
Isn't this basically just super bathsalts?

GuppyLaxx commented:
1. Horribly spammed because le meme stimulant
2. Barely used, if ever, due to how crippling it can be
3. Ends up more used as a toxin due to the fact it kills you in a tiny amount of time if ODd, if somehow  removed in time there's still a horrible addiction

XDTM commented:
You can eat 4u pills of this to get near-adminordrazine healing and stun removal for 16 seconds safely. Then again, it's hard to make and temporary so it's probably fine.

Ordonis commented:
The idea was to have a more consequence-filled version of the combat stims that syndies get and/or have a 'are you sure you want to go loud/robust?' chem. I'm more than happy to balance it in whatever direction people feel it needs balancing, but it always struck me as odd that for war ops or other confirmed threats there wasn't a chemical that 
a) rewarded chemistry for making a variety of useful medicines and not using them all to make le ebin druggs
b) was a very definite window of opportunity chem: you will NOT come off better than when you started with this, but if you need to charge headlong into bleb or other known threats, might keep you alive long enough to do something useful
c) rewards medbay for being near the frontlines with a variety of chems; combat medic always feels a little 'shove tricord and meth into them and damn the consequences, then watch as they get slaughtered' rather than 'use this stimulant and immediately prepare for the consequences that will come afterwards, having your less deadly healing chems on hand
d) punishes foolish toxmemers who think 'oh I'll just poison them with this' only to get robusted in the 15 seconds before hell begins to break loose
e) makes use of chemicals that don't necessarily get made/kept for their own purposes that much.

ma44 commented:
Inb4 odysseus syringe gun comes along. 

Tho it would probably be more far if the damage from part 1 of the addition would be moved up, as it would give the player the chance to seek out the drug instead of just immediately taking damage.

Ordonis committed:
Changes periods to commas in overdose/addiction

Fixed typo that was probably causing build to crash

Ordonis commented:
So remove the brute and oxy damage from the first stage entirely?

ma44 commented:
You can make the later tiers probably deal more damage, as the player is given a warning.

Ordonis committed:
Added in = in line 1167 that was missing

Ree spanclass warnings

Ordonis committed:
Added quotation mark on line 1167

fixing syntax errors

Tlaltecuhtli commented:
maybe give it some ingredient that cant be made in chen like gold or earthblood seems ez to make

Ordonis commented:
I'd rather add in something like gold as a catalyst rather than a full-on ingredient; cooperation is easier when you only need to give tiny amounts. I think I might go with gold or similar, simply because it turns the 'gib H2SO4' into 'trade H2SO4 for Au'.

Ordonis committed:
Fixes the rest of the improperly placed quotes+Rebalances addiction

Changes the addiction damage distribution, adds in missing quotes and
fixes indentation

Tlaltecuhtli commented:
barter gold apples for mutagen

Ordonis committed:
Further adding of = signs, changing of indentation

Skree indentation errors.

Ordonis commented:
I am absolutely certain the response to that proposed trade would be greentiding culminating with either the death of all botanists, the firebombing of botany after a dispenser was stolen, the firebombing of chemistry after a chemist makes a mistake in building a firebomb, the death of all chemists, or the delamination of the supermatter. 

R&D can't kill you too easily until later in the round, when they should have gold to spare.

Ordonis committed:
Fixes indentation of heart-attack code

Whee whitespace sensitive languages

XDTM commented:
Thing is, you _will_ come off better than you started as long as you use small doses, then rinse and repeat. Smoking this with a vape will be very powerful, as long as you let the chem wear out to reset the current_stage every once in a while.

Ordonis committed:
Another attempt at fixing inconsistent indents and proc within proc

On the tin

Ordonis committed:
More indentation work

Whitespace sensitive code why, give me back my semicolons aaaaaaaaaa

Ordonis commented:
What's the reagent distro of a vape pen? I could add in a really slow but constant brainDamage as a way to discourage constant low intensity use if you think that would improve balance

Ordonis committed:
Adds gold as a required catalyst

Exactly what it says in title

Ordonis committed:
Adds in 0 before .5, adds gold catalyst, adds braindamage

Balancing and further codefixes. Removes heart attack code because it's
too much of a pain and not likely to kill faster than the damage.

Ordonis committed:
Capitalizing A's on Adjust for stun/knock/unconscious

why is this a thing pls explain

XDTM commented:
I'm not sure what would be an ideal solution, maybe make it so it won't heal if you don't have at least 5 units. Addiction shouldn't be necessary if you do that.

WJohn commented:
This is fairly cool, I like a high risk, high reward kind of chem. Good stuff 👍

Shapsy commented:
This just sounds like it'd be a nigh-instawin if the user uses it right, and instant death if they fuck it up, which doesn't really sound fun

yorii commented:
Why not use a real chemical name? Like Modafinil for example, which is a chemical actually used by military forces worldwide today much like amphetamine was used during ww2.

Ordonis commented:
I've been told in the past to use non-realworld names for things; I was honestly thinking of calling it Pervitin (considering how nasty that ended up being) but that would skate the NSDAP rule, and I wasn't sure if Modafinil would run afoul of IP law. I'd much prefer to have a realworld or realworldish name, as opposed to NOT-KETRACEL.

Supermichael777 commented:
We need to get over the lethal ODod concept it leads to unbalanced chems that are more useful as the supertoxin of the week

Its okay to have a chem that deals constant damage(producing a second chem can also work for a lingering effect)

Ordonis committed:
Merge remote-tracking branch 'refs/remotes/tgstation/master' into New-Recipes

# Conflicts:
#	code/modules/reagents/chemistry/recipes/medicine.dm

AdamElTablawy commented:
This is pretty much a deathmix chem. The guy is strong for a little bit and then fucking dies very rapidly. 

Also fairly easy to synthesize, because m a c r o

Ordonis committed:
Fixes recipes medicine dm now that morphine is in

Headers caused code to ree

Ordonis commented:
You do need gold to synthesize it as well as heat, so macromemes will only get you partway there. Is there a general consensus that the downside needs to be less 'u dun goofd'? Because it doesn't have instakills, it is survivable if you've taken precautions ahead of time to heal yourself when the hulk-esque effect ends. It's also a means of allowing a very, very short window of opportunity for medical to help the crew fend off serious threats; it won't stop you from dying if a syndie is laying down massive amounts of fire, but you might soak a few more bullets or get close enough to interfere. Same goes with cultish threats. At the same time, the eternally pox-ridden greytide will find that they probably can't achieve much mischief with it if they manage to get their hands on it. The CMO can already make nastier deathmixes with chloral and friends.

yorii commented:
Understandable, I also realized this is nothing much at all like modafinil, I'm going to PR my own modafinil. Modafinil is also not a brand name but the actual chemical name under no actual IP.

DaedalusGame commented on code/modules/reagents/chemistry/reagents/medicine_reagents.dm:
```diff
@@ -1105,3 +1105,101 @@
+	M.AdjustUnconscious(-50*REM, 0)
+	M.adjustBrainLoss(0.5*REM,0)
+	switch(current_cycle)
+		if(13)
```
should be 0 to 13

DaedalusGame commented on code/modules/reagents/chemistry/reagents/medicine_reagents.dm:
```diff
@@ -1105,3 +1105,101 @@
+	M.AdjustUnconscious(-50*REM, 0)
+	M.adjustBrainLoss(0.5*REM,0)
+	switch(current_cycle)
+		if(13)
```
should be lower than 13 probably, or you'll get this message and then next tick your muscles begin to ache terribly

Ordonis replied:
Query: if the if(n) is set to something like 1 or 2, does that mean the positive effects will stop? That was my reason for leaving the message so late.

DaedalusGame replied:
no, the positive effects never stop because you didn't write it that way.

Ordonis replied:
Can I just add in a (low) probability check to the messages so that they appear occasionally during each stage?

DaedalusGame replied:
yepp, just do if(prob(20)) or smth

Ordonis committed:
Merge remote-tracking branch 'refs/remotes/tgstation/master' into New-Recipes

Ordonis committed:
Adds probabilities to messages to prevent spamming

See above

Ordonis committed:
Adds two missing close parens

see title

AdamElTablawy commented:
@ordonis

Three things.

1. Chloral is worthless as a deathmix chem. Never speak of it. It is forbidden.

2. Truthfully, the downsides are so significant that short of occasionally briefly popping it it’s functionally useless, because given the time and resource investment you could have made a mix with all the upsides and none of the downsides in the same timeframe, without having to worry about dying if you fuck it up.

3. As a deathmix chem this is actually very good. You can make mixes that kill faster, but considering this gives the illusion of being healthy and healing you, it likely won’t attract attention or attacks. Sure a syringe gun will, but using it as a poison or in smoke will likely incite nobody, until it’s too late. Look at how effective amantin is-it has no upsides but because it seems harmless and is mostly invisible people don’t respond even to being hyposprayed with the stuff. 

Tl;dr it’s not overpowered at all, but it actually serves better as a death chem than a combat stim since you can make better combat mixes in less time with less effort and less side effects.
